x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/fleet/components/agent_reassign_config_flyout/index.tsx

```diff
@@ -0,0 +1,186 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiSpacer,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlyoutFooter,
+  EuiSelect,
+  EuiFormRow,
+  EuiText,
+  EuiBadge,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { Datasource, Agent } from '../../../../types';
+import {
+  useGetOneAgentConfig,
+  sendPutAgentReassign,
+  useCore,
+  useGetAgentConfigs,
+} from '../../../../hooks';
+import { PackageIcon } from '../../../../components/package_icon';
+
+interface Props {
+  onClose: () => void;
+  agent: Agent;
+}
+
+export const AgentReassignConfigFlyout: React.FunctionComponent<Props> = ({ onClose, agent }) => {
+  const { notifications } = useCore();
+  const [selectedAgentConfigId, setSelectedAgentConfigId] = useState<string | undefined>(
+    agent.config_id
+  );
+
+  const agentConfigsRequest = useGetAgentConfigs();
+  const agentConfigs = agentConfigsRequest.data ? agentConfigsRequest.data.items : [];
+
+  const agentConfigRequest = useGetOneAgentConfig(selectedAgentConfigId as string);
+  const agentConfig = agentConfigRequest.data ? agentConfigRequest.data.item : null;
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  async function onSubmit() {
+    try {
+      setIsSubmitting(true);
+      if (!selectedAgentConfigId) {
+        throw new Error('No selected config id');
+      }
+      const res = await sendPutAgentReassign(agent.id, {
+        config_id: selectedAgentConfigId,
+      });
+      if (res.error) {
+        throw res.error;
+      }
+      setIsSubmitting(false);
+      const successMessage = i18n.translate(
+        'xpack.ingestManager.agentReassignConfig.successSingleNotificationTitle',
+        {
+          defaultMessage: 'Successfully changed the configuration',
+        }
+      );
+      notifications.toasts.addSuccess(successMessage);
+      onClose();
+    } catch (error) {
+      setIsSubmitting(false);
+      notifications.toasts.addError(error, {
+        title: 'Error',
+      });
+    }
+  }
+
+  return (
+    <EuiFlyout onClose={onClose} size="l" maxWidth={640}>
+      <EuiFlyoutHeader hasBorder aria-labelledby="FleetAgentEnrollmentFlyoutTitle">
+        <EuiTitle size="m">
+          <h2 id="FleetAgentReassigmentFlyoutTitle">
+            <FormattedMessage
+              id="xpack.ingestManager.agentReassignConfig.flyoutTitle"
+              defaultMessage="Assign new agent configuration"
+            />
+          </h2>
+        </EuiTitle>
+        <EuiSpacer size="m" />
+        <EuiText size="s">
+          <FormattedMessage
+            id="xpack.ingestManager.agentReassignConfig.flyoutDescription"
+            defaultMessage="Choose a new agent configuration to assign the selected agent(s) to."
```

Comment:
```suggestion
            defaultMessage="Choose a new agent configuration to assign the selected agent to."
```

(since we are not supporting bulk at the moment)